core/src/util/AndroidCompat.cpp

```diff
@@ -0,0 +1,49 @@
+/*=============================================================================
+
+ Library: CppMicroServices
+
+ Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+ file at the top-level directory of this distribution and at
+ https://github.com/saschazelzer/CppMicroServices/COPYRIGHT .
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ =============================================================================*/
+
+#if defined(__ANDROID__)
+
+#include "AndroidCompat.h"
+
+#include "usAny.h"
+
+namespace {
+template <typename T>
+std::string internal_to_string(T val)
+{
+  return us::Any(val).ToString();
```

Comment:
I think we should just use a `std::stringstream` here directly instead of going through `us::Any`. The any class uses a string stream as well, but it also does a dynamic memory allocation on construction.